src/main/java/frc/team88/swerve/configuration/Configuration.java

```diff
@@ -260,6 +262,27 @@ private NavX instantiateNavX(Config gyroConfig) {
     }
   }
 
+  /**
+   * Instatiates a Pigeon object.
+   *
+   * @param gyroConfig The config for the gyro.
+   * @return The Pigeon object.
+   * @throws SwerveConfigException If the user provided config is incorrect.
+   */
+  private Pigeon instantiatePigeon(Config gyroConfig) {
+    String portType = configCheckAndGet(gyroConfig, "port-type", String.class);
+    switch (portType) {
+      case "TalonSRX":
```

Comment:
I'm not too familiar with the Pigeon, but is the idea that it would be connected to a Talon SRX that is also being used to control a non-swerve motor? My concern is that as you currently have this written, there would be no way to access the TalonSRX object to use it outside of this library. My initial thoughts are to do one of 2 things:

* Add an interface to get the TalonSRX out of the library via SwerveController, similar to what is done with CANifiers
* Leave this the way it is, so that it if it is specified via the config the Talon can't be used outside the library, and let teams pass a Pigeon object into the SwerveController constructor instead of using the config if they need access to the TalonSRX

Reply:
I think we leave it